index.js

```diff
@@ -198,14 +212,17 @@ URL.prototype.set = function set(part, value, fn) {
     url[part] = value;
 
     if (!required(value, url.protocol)) {
-      url.host = url.hostname;
+      url.host = isIPv6(url.hostname) ? '['+ url.hostname +']' : url.hostname;
       url[part] = '';
     } else if (value) {
-      url.host = url.hostname +':'+ value;
+      url.host = isIPv6(url.hostname) ? '['+ url.hostname +']' +':'+ value : url.hostname +':'+ value;
```

Comment:
Yes, it probably makes sense to use a tmp variable, I would simply call it `host`.